src/AWS.Deploy.Common/TypeHintData/TypeHintResourceColumn.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+namespace AWS.Deploy.Common.TypeHintData
+{
+    /// <summary>
+    /// Represents the column header for a list/grid of <see cref="TypeHintResource"/> rows
+    /// </summary>
+    public class TypeHintResourceColumn
+    {
+        /// <summary>
+        /// Internal Id for the column. This can be used as a key to access a specific
+        /// row's value via its <see cref="TypeHintResource.ColumnValues"/>
+        /// </summary>
+        public string Id { get; set; }
```

Comment:
I liked the keys for:
* Readability, since `colums['foo']` describes which column you're accessing for a given row
* If we ever reorder columns, there are fewer code changes to make

But I was also slightly afraid of us mistyping a key and then getting a `KeyNotFound ` at runtime.

----

Dropped the Ids and dictionaries for now, and switched to lists with indices.